src/compas_timber/fabrication/btlx.py

```diff
@@ -51,113 +43,166 @@ class BTLx(object):
+        :class:`~xml.etree.ElementTree.Element`
+            The file history element.
+
+        """
+        # create file history element
         file_history = ET.Element("FileHistory")
-        file_history.append(ET.Element("InitialExportProgram", self.history))
+        # create initial export program element
+        file_history.append(ET.Element("InitialExportProgram", cls.FILE_HISTORY_ATTRIBUTES))
         return file_history
 
+    @classmethod
+    def _create_project_element(cls, model):
+        """Creates the project element. This method creates the parts element and appends it to the project element.
+
+        Returns
+        -------
+        :class:`~xml.etree.ElementTree.Element`
+            The project element.
+
+        """
+        # create project element
+        project_element = ET.Element(
+            "Project", Name="testProject"
+        )  # TODO: Should the name be set from the model and passed here?
+        # create parts element
+        parts_element = ET.SubElement(project_element, "Parts")
+        # create part elements for each beam
+        for i, beam in enumerate(model.beams):
+            part_element = cls._create_part(beam, i)
+            parts_element.append(part_element)
+        return project_element
+
+    @classmethod
+    def _create_part(cls, beam, order_num):
+        """Creates a part element. This method creates the processing elements and appends them to the part element.
+
+        Parameters
+        ----------
+        beam : :class:`~compas_timber.elements.Beam`
+            The beam object.
+        num : int
+            The order number of the part.
+
+        Returns
+        -------
+        :class:`~xml.etree.ElementTree.Element`
+            The part element.
+
+        """
+        # create part element
+        part = BTLxPart(beam, order_num=order_num)
+        part_element = ET.Element("Part", part.attr)
+        part_element.extend([part.et_transformations, part.et_grain_direction, part.et_reference_side])
+        # create processings element for the part
+        processings_element = ET.Element("Processings")
+        for feature in beam.features:
+            processing_element = cls._create_processing(feature)
+            processings_element.append(processing_element)
+        part_element.extend([processings_element, part.et_shape])
+        return part_element
+
+    @classmethod
+    def _create_processing(cls, processing):
+        """Creates a processing element. This method creates the subprocess elements and appends them to the processing element.
+
+        Parameters
+        ----------
+        processing : :class:`~compas_timber.fabrication.btlx.BTLxProcessing`
+            The processing object.
+
+        Returns
+        -------
+        :class:`~xml.etree.ElementTree.Element`
+            The processing element.
+
+        """
+        # create processing element
+        processing_element = ET.Element(
+            processing.PROCESSING_NAME,
```

Comment:
we still have a mix of the old `Features` and the new `Processing`, this will break here, since `Features` don't have `PROCESSING_NAME` (or any other useful things). while we deal with deprecating the old system, maybe we check somewhere here for these old features and just ignore them for now. that way BTLX won't fail if there's a `MillVolume` somewhere (processing will be missing of course, but will be introduced in the future)